frontend/src/components/HomePage/home.jsx

```diff
@@ -233,7 +229,14 @@ export default function Home({ selectedEvent, setEvent }) {
             />
           </Route>
           <Route path="/waiver">
-            <WaiverPage user={user} />
+            {user.signedWaiver ? (
+              <RegistrationComplete
+                selectedEvent={selectedEvent}
+                handleModalClose={handleModalClose}
+              />
+            ) : (
+              <WaiverPage user={user} />
+            )}
```

Comment:
Could you change where the link on the SignUp page goes instead of changing this route dynamically? Typically it's not the best practice to have the route change dynamically like this.